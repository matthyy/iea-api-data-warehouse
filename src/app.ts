import express from 'express'
import health from './routes/health_router'
import coupon from './routes'
import { createConnection } from './lib/db_connection'
import { errorHandler } from './middleware/error_handler'
import helmet from 'helmet'

createConnection()
export const app = express()
app.use(express.json())
app.use(helmet())

app.use(health)
app.use(coupon)

app.use(errorHandler)
