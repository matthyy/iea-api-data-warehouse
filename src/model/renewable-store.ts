import mongoose, { Schema } from 'mongoose'

export type RenewableStore = {
    country: string
    year: number
    nbVehicle: number
    nbCharger: number
}

export type RenewableAggregate = {
    _id: string
    totalVehicle: number
    totalCharger: number
}

export type RenewableStoreDoc = RenewableStore & mongoose.Document<mongoose.Types.ObjectId>

export const RenewableStoreSchema = new mongoose.Schema<RenewableStoreDoc>({
    country: { type: Schema.Types.String, required: true },
    year: { type: Schema.Types.Number, required: true },
    nbVehicle: { type: Schema.Types.Number, required: true },
    nbCharger: { type: Schema.Types.Number, required: true },
})

RenewableStoreSchema.index({ country: 1, year: 1 })

export const RenewableStoreModel = mongoose.model<RenewableStoreDoc>('RenewableStore', RenewableStoreSchema)
