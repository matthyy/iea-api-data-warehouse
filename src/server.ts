import { app } from './app'
import { logger } from './lib/logger'

const port = process.env.PORT || 3000

app.listen(port, () => {
    logger.info(`Coupon API Listening on port ${port}`)
})
