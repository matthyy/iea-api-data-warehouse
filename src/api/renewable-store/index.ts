import { NextFunction, Request, Response } from 'express'
import { fetchRenewableData } from '../../services'

/**
 * req.params should be validate with validator schema
 */
export const getRenewableData = async (
    req: Request<
        undefined,
        undefined,
        undefined,
        { from: string; to: string; country?: string; year?: number; sort?: 'asc' | 'desc' }
    >,
    res: Response,
    next: NextFunction
): Promise<void> => {
    try {
        const vehicles = await fetchRenewableData(req.query)

        res.status(200).json(vehicles)
    } catch (error) {
        next(error)
    }
}
