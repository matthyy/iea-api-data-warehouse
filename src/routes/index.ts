import { Router } from 'express'
import { getRenewableData } from '../api/renewable-store'
import { checkValidQuery } from '../middleware/validation'

const router = Router({ mergeParams: true })

router.get('/renewableData', checkValidQuery, getRenewableData)
export default router
