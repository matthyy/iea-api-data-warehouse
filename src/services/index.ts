import { RenewableAggregate, RenewableStore, RenewableStoreModel } from '../model/renewable-store'
import { FilterQuery } from 'mongoose'

export const fetchRenewableData = async ({
    country,
    from,
    to,
    year,
    sort,
}: {
    from?: string
    to?: string
    country?: string
    year?: number
    sort?: 'asc' | 'desc'
}): Promise<{
    aggregate: RenewableAggregate
    results: RenewableStore[]
}> => {
    const filter: FilterQuery<RenewableStore> = {}
    const match: { country?: string; year?: number } = {}
    if (country) {
        filter.country = country
        match.country = country
    }

    if (year) {
        filter.year = year
        match.year = year
    }

    const aggregate = []
    aggregate.push({ $match: match })

    if (from && to) {
        filter.year = {
            $gt: parseInt(from),
            $lt: parseInt(to),
        }
        aggregate.push({
            $match: {
                year: {
                    $gt: from ? parseInt(from) : 2000,
                    $lt: to ? parseInt(to) : 2021,
                },
            },
        })
    }

    aggregate.push({
        $group: { _id: country || 'all', totalVehicle: { $sum: '$nbVehicle' }, totalCharger: { $sum: '$nbCharger' } },
    })

    const [count, results] = await Promise.all([
        RenewableStoreModel.aggregate<RenewableAggregate>(aggregate),
        RenewableStoreModel.find(filter)
            .sort({ year: sort === 'asc' ? 1 : -1 })
            .lean()
            .exec(),
    ])

    return {
        aggregate: count[0],
        results,
    }
}
