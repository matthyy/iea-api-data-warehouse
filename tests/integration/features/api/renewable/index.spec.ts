import supertest from 'supertest'
import mongoose from 'mongoose'
import { app } from '../../../../../src/app'
import { RenewableAggregate, RenewableStore, RenewableStoreModel } from '../../../../../src/model/renewable-store'
import features from './features/index.json'

const request = supertest(app)

describe('API Renewable Store', () => {
    beforeAll(async () => {
        await RenewableStoreModel.insertMany(features)
    })

    afterAll(async () => {
        await RenewableStoreModel.deleteMany().exec()
        await mongoose.disconnect()
    })

    test('should return renewableData', async () => {
        const response = await request.get('/renewableData').expect(200)
        const snapshot = response.body as {
            aggregate: RenewableAggregate
            results: RenewableStore[]
        }
        snapshot.results = helperFormatSnapshot(snapshot.results)

        expect(snapshot).toMatchSnapshot()
    })
    test('should return renewableData with country queryParam set', async () => {
        const response = await request.get('/renewableData?country=australia').expect(200)
        const snapshot = response.body as {
            aggregate: RenewableAggregate
            results: RenewableStore[]
        }
        snapshot.results = helperFormatSnapshot(snapshot.results)

        expect(snapshot).toMatchSnapshot()
    })

    test('should return renewableData with country queryParam from and to set', async () => {
        const response = await request.get('/renewableData?from=2000&to=2019').expect(200)
        const snapshot = response.body as {
            aggregate: RenewableAggregate
            results: RenewableStore[]
        }
        snapshot.results = helperFormatSnapshot(snapshot.results)

        expect(snapshot).toMatchSnapshot()
    })

    test('should return renewableData with country queryParam from, to and country set', async () => {
        const response = await request.get('/renewableData?from=2000&to=2019&country=australia').expect(200)
        const snapshot = response.body as {
            aggregate: RenewableAggregate
            results: RenewableStore[]
        }
        snapshot.results = helperFormatSnapshot(snapshot.results)

        expect(snapshot).toMatchSnapshot()
    })

    test('should return renewableData with country queryParam sort set', async () => {
        const response = await request.get('/renewableData?sort=asc').expect(200)
        const snapshot = response.body as {
            aggregate: RenewableAggregate
            results: RenewableStore[]
        }
        snapshot.results = helperFormatSnapshot(snapshot.results)

        expect(snapshot).toMatchSnapshot()
    })

    test('should return renewableData with country queryParam year set', async () => {
        const response = await request.get('/renewableData?year=2019').expect(200)
        const snapshot = response.body as {
            aggregate: RenewableAggregate
            results: RenewableStore[]
        }
        snapshot.results = helperFormatSnapshot(snapshot.results)

        expect(snapshot).toMatchSnapshot()
    })
})

const helperFormatSnapshot = (list: RenewableStore[]): RenewableStore[] => {
    return list.map((result) => ({
        year: result.year,
        nbVehicle: result.nbVehicle,
        nbCharger: result.nbCharger,
        country: result.country,
    }))
}
