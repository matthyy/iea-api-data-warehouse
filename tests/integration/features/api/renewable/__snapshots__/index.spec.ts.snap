// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API Renewable Store should return renewableData 1`] = `
Object {
  "aggregate": Object {
    "_id": "all",
    "totalCharger": 3605.62,
    "totalVehicle": 27.89,
  },
  "results": Array [
    Object {
      "country": "australia",
      "nbCharger": 1930,
      "nbVehicle": 8.6,
      "year": 2019,
    },
    Object {
      "country": "brazil",
      "nbCharger": 13.3,
      "nbVehicle": 2.08,
      "year": 2019,
    },
    Object {
      "country": "australia",
      "nbCharger": 727.02,
      "nbVehicle": 5.72,
      "year": 2018,
    },
    Object {
      "country": "brazil",
      "nbCharger": 459.3,
      "nbVehicle": 0.71,
      "year": 2018,
    },
    Object {
      "country": "australia",
      "nbCharger": 476,
      "nbVehicle": 3.92,
      "year": 2017,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0.36,
      "year": 2017,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 2.85,
      "year": 2016,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0.08,
      "year": 2016,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 2.15,
      "year": 2015,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0.03,
      "year": 2015,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 1.13,
      "year": 2014,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2014,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0.18,
      "year": 2013,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2013,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0.08,
      "year": 2012,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2012,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2011,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2011,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2010,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2010,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2009,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2009,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2008,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2008,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2007,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2007,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2006,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2006,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2005,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2005,
    },
  ],
}
`;

exports[`API Renewable Store should return renewableData with country queryParam from and to set 1`] = `
Object {
  "aggregate": Object {
    "_id": "all",
    "totalCharger": 1662.32,
    "totalVehicle": 17.21,
  },
  "results": Array [
    Object {
      "country": "australia",
      "nbCharger": 727.02,
      "nbVehicle": 5.72,
      "year": 2018,
    },
    Object {
      "country": "brazil",
      "nbCharger": 459.3,
      "nbVehicle": 0.71,
      "year": 2018,
    },
    Object {
      "country": "australia",
      "nbCharger": 476,
      "nbVehicle": 3.92,
      "year": 2017,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0.36,
      "year": 2017,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 2.85,
      "year": 2016,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0.08,
      "year": 2016,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 2.15,
      "year": 2015,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0.03,
      "year": 2015,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 1.13,
      "year": 2014,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2014,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0.18,
      "year": 2013,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2013,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0.08,
      "year": 2012,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2012,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2011,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2011,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2010,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2010,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2009,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2009,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2008,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2008,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2007,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2007,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2006,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2006,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2005,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2005,
    },
  ],
}
`;

exports[`API Renewable Store should return renewableData with country queryParam from, to and country set 1`] = `
Object {
  "aggregate": Object {
    "_id": "australia",
    "totalCharger": 1203.02,
    "totalVehicle": 16.03,
  },
  "results": Array [
    Object {
      "country": "australia",
      "nbCharger": 727.02,
      "nbVehicle": 5.72,
      "year": 2018,
    },
    Object {
      "country": "australia",
      "nbCharger": 476,
      "nbVehicle": 3.92,
      "year": 2017,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 2.85,
      "year": 2016,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 2.15,
      "year": 2015,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 1.13,
      "year": 2014,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0.18,
      "year": 2013,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0.08,
      "year": 2012,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2011,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2010,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2009,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2008,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2007,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2006,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2005,
    },
  ],
}
`;

exports[`API Renewable Store should return renewableData with country queryParam set 1`] = `
Object {
  "aggregate": Object {
    "_id": "australia",
    "totalCharger": 3133.02,
    "totalVehicle": 24.63,
  },
  "results": Array [
    Object {
      "country": "australia",
      "nbCharger": 1930,
      "nbVehicle": 8.6,
      "year": 2019,
    },
    Object {
      "country": "australia",
      "nbCharger": 727.02,
      "nbVehicle": 5.72,
      "year": 2018,
    },
    Object {
      "country": "australia",
      "nbCharger": 476,
      "nbVehicle": 3.92,
      "year": 2017,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 2.85,
      "year": 2016,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 2.15,
      "year": 2015,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 1.13,
      "year": 2014,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0.18,
      "year": 2013,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0.08,
      "year": 2012,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2011,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2010,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2009,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2008,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2007,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2006,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2005,
    },
  ],
}
`;

exports[`API Renewable Store should return renewableData with country queryParam sort set 1`] = `
Object {
  "aggregate": Object {
    "_id": "all",
    "totalCharger": 3605.62,
    "totalVehicle": 27.89,
  },
  "results": Array [
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2005,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2005,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2006,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2006,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2007,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2007,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2008,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2008,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2009,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2009,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2010,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2010,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2011,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2011,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0.08,
      "year": 2012,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2012,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 0.18,
      "year": 2013,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2013,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 1.13,
      "year": 2014,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0,
      "year": 2014,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 2.15,
      "year": 2015,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0.03,
      "year": 2015,
    },
    Object {
      "country": "australia",
      "nbCharger": 0,
      "nbVehicle": 2.85,
      "year": 2016,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0.08,
      "year": 2016,
    },
    Object {
      "country": "australia",
      "nbCharger": 476,
      "nbVehicle": 3.92,
      "year": 2017,
    },
    Object {
      "country": "brazil",
      "nbCharger": 0,
      "nbVehicle": 0.36,
      "year": 2017,
    },
    Object {
      "country": "australia",
      "nbCharger": 727.02,
      "nbVehicle": 5.72,
      "year": 2018,
    },
    Object {
      "country": "brazil",
      "nbCharger": 459.3,
      "nbVehicle": 0.71,
      "year": 2018,
    },
    Object {
      "country": "australia",
      "nbCharger": 1930,
      "nbVehicle": 8.6,
      "year": 2019,
    },
    Object {
      "country": "brazil",
      "nbCharger": 13.3,
      "nbVehicle": 2.08,
      "year": 2019,
    },
  ],
}
`;

exports[`API Renewable Store should return renewableData with country queryParam year set 1`] = `
Object {
  "aggregate": Object {
    "_id": "all",
    "totalCharger": 1943.3,
    "totalVehicle": 10.68,
  },
  "results": Array [
    Object {
      "country": "australia",
      "nbCharger": 1930,
      "nbVehicle": 8.6,
      "year": 2019,
    },
    Object {
      "country": "brazil",
      "nbCharger": 13.3,
      "nbVehicle": 2.08,
      "year": 2019,
    },
  ],
}
`;
